File: generate_mermaid.php
<?php

$sql = file_get_contents('schema_only.sql');

preg_match_all('/CREATE TABLE `(\w+)` \((.*?)\) ENGINE/s', $sql, $tables);
preg_match_all('/ALTER TABLE `(\w+)`\s+ADD CONSTRAINT `.*?` FOREIGN KEY \(`(\w+)`\) REFERENCES `(\w+)` \(`(\w+)`\)/', $sql, $fks, PREG_SET_ORDER);

$mermaid = "erDiagram\n";

for ($i = 0; $i < count($tables[0]); $i++) {
    $table = $tables[1][$i];
    $cols = $tables[2][$i];
    
    $mermaid .= "    {$table} {\n";
    $lines = explode("\n", $cols);
    foreach ($lines as $line) {
        $line = trim($line);
        if (strpos($line, '`') === 0) {
            $parts = explode(' ', $line);
            $colName = str_replace('`', '', $parts[0]);
            $colType = $parts[1];
            if (strpos($colType, '(') !== false) {
                $colType = explode('(', $colType)[0];
            }
            $mermaid .= "        {$colType} {$colName}\n";
        }
    }
    $mermaid .= "    }\n\n";
}

foreach ($fks as $fk) {
    $table = $fk[1];
    $col = $fk[2];
    $refTable = $fk[3];
    $refCol = $fk[4];
    $mermaid .= "    {$refTable} ||--o{ {$table} : \"{$col}\"\n";
}

file_put_contents('mermaid.txt', $mermaid);
echo "Mermaid generated.";

File: generate_mermaid.py
import re

with open('schema_only.sql', 'r', encoding='utf-8') as f:
    sql = f.read()

# Extract tables
tables = re.findall(r'CREATE TABLE `(\w+)` \((.*?)\) ENGINE', sql, re.DOTALL)
fks = re.findall(r'ALTER TABLE `(\w+)`\s+ADD CONSTRAINT `.*?` FOREIGN KEY \(`(\w+)`\) REFERENCES `(\w+)` \(`(\w+)`\)', sql)

mermaid = "erDiagram\n"

for table, cols in tables:
    mermaid += f"    {table} {{\n"
    # parse cols
    lines = cols.split('\n')
    for line in lines:
        line = line.strip()
        if line.startswith('`'):
            parts = line.split(' ')
            col_name = parts[0].replace('`', '')
            col_type = parts[1]
            if '(' in col_type:
                col_type = col_type.split('(')[0]
            mermaid += f"        {col_type} {col_name}\n"
    mermaid += "    }\n\n"

for fk in fks:
    table, col, ref_table, ref_col = fk
    mermaid += f"    {ref_table} ||--o{{ {table} : \"{col}\"\n"

with open('mermaid.txt', 'w', encoding='utf-8') as f:
    f.write(mermaid)
print("Mermaid generated.")
